Templates/Common-Backend-Scripts/wazideploy-generate.sh

```diff
@@ -358,23 +458,70 @@ validateOptions() {
   if [ ! -z "${ConfigFile}" ]; then
     if [ ! -f "${ConfigFile}" ]; then
       rc=8
-      ERRMSG=$PGM": [ERROR] Specified Wazi Deploy Artifactory Configuration file (${ConfigFile}) was not found. rc="$rc
+      ERRMSG=$PGM": [ERROR] Specified Wazi Deploy Artifact repository configuration file (${ConfigFile}) was not found. rc="$rc
       echo $ERRMSG
     fi
   fi
 
   # compute the output file
   if [ -z "${PackageOutputFile}" ] && [ ! -z "${ConfigFile}" ]; then
-    # c
     checkWorkspace
     PackageOutputFile="$(wdDeployPackageDir)"
   fi
   # if relative path
-  if [[ ! ${PackageOutputFile:0:1} == "/" ]] && [[ ! -z "${PackageOutputFile}" ]] ; then
+  if [[ ! ${PackageOutputFile:0:1} == "/" ]] && [[ ! -z "${PackageOutputFile}" ]]; then
     PackageOutputFile="$(wdDeployPackageDir)/${PackageOutputFile}"
   fi
 }
 
+# When publishing is enabled, try reading the wdPackageVersionFile
+# that needs to be computed before this step.
+if [ $rc -eq 0 ] && [ "$publish" == "true" ] && [ ! -z "${buildIdentifier}" ]; then
+  checkWorkspace
+  CMD="${computePackageUrlUtil} -w $Workspace -a $App -b $Branch -i $buildIdentifier"
+
+  if [ ! -z "${PipelineType}" ]; then
+    CMD+=" -p ${PipelineType}"
+  else
+    rc=8
+    ERRMSG=$PGM": [ERROR] To compute the Package Url to automatically download the tar file via Wazi Deploy generate, you need to provide the pipelineType. rc="$rc
+    echo $ERRMSG
+  fi
+
+  if [ ! -z "${releaseIdentifier}" ]; then
+    CMD+=" -r ${releaseIdentifier}"
+  fi
+
+  if [ $rc -eq 0 ]; then
+    echo $PGM": [INFO] ** Compute Package Url based on existing conventions using command"
+    echo $PGM": [INFO] ** Invoking subtask ${CMD}"
+    ${CMD}
+    rc=$?
+    if [ $rc -eq 0 ]; then
+      if [ -f "$(getLogDir)/${wdPackageVersionFile}" ]; then
+        echo $PGM": [INFO] ** Read configuration file $(getLogDir)/${wdPackageVersionFile}"
+        source "$(getLogDir)/${wdPackageVersionFile}"
+      else
+        rc=4
+        ERRMSG=$PGM": [ERROR] ** The configuration file $(getLogDir)/${wdPackageVersionFile} was not found. Check previous console output. rc="$rc
+        echo $ERRMSG
+      fi
+    fi
+
+    # When a the packageUrl environment variable found in the file continue to compute
+    if [ $rc -eq 0 ] && [ ! -z "${packageUrl}" ]; then
+      echo $PGM": [INFO] ** Package Url configuration file found. Package Input File will be set to ${packageUrl}. Package Output file will be computed."
+      PackageInputFile="${packageUrl}"
+      ## Take the last segment of the URL to define the tarFileName
+      tarFileName=$(echo $PackageInputFile | awk -F "/" '{print $NF}')
+      PackageOutputFile="$(wdDeployPackageDir)/${tarFileName}"
+      echo $PGM": [INFO] ** Package Output file information stored in $(getLogDir)/${wdPackageVersionFile}."
+      echo "PackageInputFile=${PackageOutputFile}" >> $(getLogDir)/${wdPackageVersionFile}
```

Comment:
We should avoid using the same file to convey the local path of the archive to deploy.
Why not using a fixed name. We could create a subfolder `toBeDeployed` in $wdDeployPackageDir, and we store the package to be deployed by WD Deploy in there, so we know it's theoretically always there.

Reply:
This is addressed.